File: primo-anno/Programmazione/C/esame_160626.c
#include <stdio.h>
#include <stdlib.h>


int *verifica_sequenziale(int *a, int na, int *t, int nt){

    int j = 0; 

    int *p = malloc(sizeof(int)*nt); 
    if(p == NULL){
        return NULL; 
    }

    for(int i = 0; i < na; i++){

        if (j < nt && a[i] == t[j]){
            p[j] = i; 
            j++;
        }

    }
    
    if(j != nt){
        return NULL;
        }

    return p; 

}

int main(){

    int a[] = {33, 43, 35, 31, 50, 49, 35, 37, 50, 100}; 
    int na = sizeof(a)/sizeof(int); 
    int t[] = {43, 35, 31, 49, 35, 50}; 
    int nt = sizeof(t)/sizeof(int); 

    int *res = verifica_sequenziale(a, na, t, nt); 
    for(int i = 0; i < nt; i++){
        printf("%d \n", res[i]); 
    }

}

/*complessità temporale O(n), spaziale O(n)*/

File: primo-anno/Programmazione/python/esame_160626.py
'''siano date due liste di interi a e t, a rappresenta una lista di codici numerici, t una lista con codici target
di cui si vuole verificare la presenza e l'ordine in a. Scrivere una funzione python chiamata verifica_sequenza(a, t)
che nel caso in cui la lista a contenga tutti gli elementi di t nell'esatto ordine in cui compaiono in t, 
restituisca una lista p dove p[i] indica la posizione in a in cui compare t[i]
altrimenti restituisca la lista vuota '''

def verifica_sequenziale(a, t):

    p = []
    j = 0
    n = len(t)
    

    for i in range(len(a)):

        if  j < n and a[i] == t[j]: 
            p.append(i)
            j += 1
        
    if len(p) != len(t): 
        return []

    

    return p
         
    





a =[33, 43, 35, 31, 50, 49, 35, 37, 50, 100]
t = [43, 35, 31, 49, 35, 50]


print(verifica_sequenziale(a, t))

#complessità spaziale 